### ISSUE TITLE
Could not pass event PlayerDeathEvent to GriefPrevention v16.18

### ISSUE THREAD

**Issue by Brasil150** [GriefPrevention/GriefPrevention#2193]
### Observed Behavior

Error messages in red at some seemingly random times in version 1.20.4

### Expected Behavior

No apparent errors.

### Reproduction steps

1. Use Purpur 1.20.4
2. Start the server and play for some time.

### Stack trace or error log

```shell
[22:16:48 ERROR]: Could not pass event PlayerDeathEvent to GriefPrevention v16.18
java.lang.IllegalArgumentException: message cannot be null
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143) ~[guava-32.1.2-jre.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendRawMessage(CraftPlayer.java:331) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendRawMessage(CraftPlayer.java:326) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendMessage(CraftPlayer.java:343) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at me.ryanhamshire.GriefPrevention.PlayerEventHandler.onPlayerDeath(PlayerEventHandler.java:894) ~[GriefPrevention.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:git-Purpur-2119]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.callPlayerDeathEvent(CraftEventFactory.java:1000) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.level.ServerPlayer.die(ServerPlayer.java:987) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.hurt(LivingEntity.java:1603) ~[?:?]
	at net.minecraft.world.entity.player.Player.hurt(Player.java:1007) ~[?:?]
	at net.minecraft.server.level.ServerPlayer.hurt(ServerPlayer.java:1165) ~[?:?]
	at net.minecraft.world.entity.Mob.doHurtTarget(Mob.java:1722) ~[?:?]
	at net.minecraft.world.entity.monster.Blaze$BlazeAttackGoal.tick(Blaze.java:253) ~[?:?]
	at net.minecraft.world.entity.ai.goal.WrappedGoal.tick(WrappedGoal.java:63) ~[?:?]
	at net.minecraft.world.entity.ai.goal.GoalSelector.tickRunningGoals(GoalSelector.java:157) ~[?:?]
	at net.minecraft.world.entity.ai.goal.GoalSelector.tick(GoalSelector.java:148) ~[?:?]
	at net.minecraft.world.entity.Mob.serverAiStep(Mob.java:960) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3485) ~[?:?]
	at net.minecraft.world.entity.Mob.aiStep(Mob.java:706) ~[?:?]
	at net.minecraft.world.entity.monster.Monster.aiStep(Monster.java:45) ~[?:?]
	at net.minecraft.world.entity.monster.Blaze.aiStep(Blaze.java:146) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3134) ~[?:?]
	at net.minecraft.world.entity.Mob.tick(Mob.java:462) ~[?:?]
	at net.minecraft.server.level.ServerLevel.tickNonPassenger(ServerLevel.java:1498) ~[?:?]
	at net.minecraft.server.level.ServerLevel.lambda$tick$8(ServerLevel.java:946) ~[?:?]
	at net.minecraft.world.level.entity.EntityTickList.forEach(EntityTickList.java:42) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:923) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1746) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:486) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1545) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1235) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:326) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[22:16:55 INFO]: PeweeOfi issued server command: /back
[22:17:04 ERROR]: Could not pass event PlayerDeathEvent to GriefPrevention v16.18
java.lang.IllegalArgumentException: message cannot be null
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143) ~[guava-32.1.2-jre.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendRawMessage(CraftPlayer.java:331) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendRawMessage(CraftPlayer.java:326) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendMessage(CraftPlayer.java:343) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at me.ryanhamshire.GriefPrevention.PlayerEventHandler.onPlayerDeath(PlayerEventHandler.java:894) ~[GriefPrevention.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:git-Purpur-2119]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.callPlayerDeathEvent(CraftEventFactory.java:1000) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.level.ServerPlayer.die(ServerPlayer.java:987) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.hurt(LivingEntity.java:1603) ~[?:?]
	at net.minecraft.world.entity.player.Player.hurt(Player.java:1007) ~[?:?]
	at net.minecraft.server.level.ServerPlayer.hurt(ServerPlayer.java:1165) ~[?:?]
	at net.minecraft.world.effect.WitherMobEffect.applyEffectTick(WitherMobEffect.java:13) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.world.effect.MobEffectInstance.tick(MobEffectInstance.java:218) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.tickEffects(LivingEntity.java:930) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.baseTick(LivingEntity.java:509) ~[?:?]
	at net.minecraft.world.entity.Entity.tick(Entity.java:841) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3093) ~[?:?]
	at net.minecraft.world.entity.player.Player.tick(Player.java:301) ~[?:?]
	at net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:789) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:353) ~[?:?]
	at net.minecraft.network.Connection.tick(Connection.java:601) ~[?:?]
	at net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:240) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1776) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:486) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1545) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1235) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:326) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[22:17:11 INFO]: PeweeOfi issued server command: /back
[22:17:18 ERROR]: Could not pass event PlayerDeathEvent to GriefPrevention v16.18
java.lang.IllegalArgumentException: message cannot be null
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:143) ~[guava-32.1.2-jre.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendRawMessage(CraftPlayer.java:331) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendRawMessage(CraftPlayer.java:326) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.craftbukkit.v1_20_R3.entity.CraftPlayer.sendMessage(CraftPlayer.java:343) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at me.ryanhamshire.GriefPrevention.PlayerEventHandler.onPlayerDeath(PlayerEventHandler.java:894) ~[GriefPrevention.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:git-Purpur-2119]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[purpur-api-1.20.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R3.event.CraftEventFactory.callPlayerDeathEvent(CraftEventFactory.java:1000) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.level.ServerPlayer.die(ServerPlayer.java:987) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.hurt(LivingEntity.java:1603) ~[?:?]
	at net.minecraft.world.entity.player.Player.hurt(Player.java:1007) ~[?:?]
	at net.minecraft.server.level.ServerPlayer.hurt(ServerPlayer.java:1165) ~[?:?]
	at net.minecraft.world.entity.Mob.doHurtTarget(Mob.java:1722) ~[?:?]
	at net.minecraft.world.entity.monster.WitherSkeleton.doHurtTarget(WitherSkeleton.java:142) ~[?:?]
	at net.minecraft.world.entity.ai.goal.MeleeAttackGoal.checkAndPerformAttack(MeleeAttackGoal.java:129) ~[?:?]
	at net.minecraft.world.entity.ai.goal.MeleeAttackGoal.tick(MeleeAttackGoal.java:121) ~[?:?]
	at net.minecraft.world.entity.ai.goal.WrappedGoal.tick(WrappedGoal.java:63) ~[?:?]
	at net.minecraft.world.entity.ai.goal.GoalSelector.tickRunningGoals(GoalSelector.java:157) ~[?:?]
	at net.minecraft.world.entity.ai.goal.GoalSelector.tick(GoalSelector.java:148) ~[?:?]
	at net.minecraft.world.entity.Mob.serverAiStep(Mob.java:960) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3485) ~[?:?]
	at net.minecraft.world.entity.Mob.aiStep(Mob.java:706) ~[?:?]
	at net.minecraft.world.entity.monster.Monster.aiStep(Monster.java:45) ~[?:?]
	at net.minecraft.world.entity.monster.AbstractSkeleton.aiStep(AbstractSkeleton.java:113) ~[?:?]
	at net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:3134) ~[?:?]
	at net.minecraft.world.entity.Mob.tick(Mob.java:462) ~[?:?]
	at net.minecraft.server.level.ServerLevel.tickNonPassenger(ServerLevel.java:1498) ~[?:?]
	at net.minecraft.server.level.ServerLevel.lambda$tick$8(ServerLevel.java:946) ~[?:?]
	at net.minecraft.world.level.entity.EntityTickList.forEach(EntityTickList.java:42) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:923) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1746) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:486) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1545) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1235) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:326) ~[purpur-1.20.4.jar:git-Purpur-2119]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### Server version

```shell
[22:26:01 INFO]: Current: git-Purpur-2119 (MC: 1.20.4)*
* You are 1 version(s) behind
```

### GriefPrevention version

```shell
GriefPrevention 16.18.1
```

### Configuration

```yaml
# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/

GriefPrevention:
  SeaLevelOverrides:
    world: -1
    world_nether: -1
    world_the_end: -1
    spawn: -1
  Claims:
    Mode:
      world: Survival
      world_the_end: Survival
      world_nether: Survival
      spawn: Disabled
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 100
    Claim Blocks Accrued Per Hour:
      Default: 100
    Max Accrued Claim Blocks:
      Default: 80000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 10
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 5
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: -2147483648
    InvestigationTool: STICK
    ModificationTool: GOLDEN_SHOVEL
    Expiration:
      ChestClaimDays: 7
      UnusedClaimDays: 14
      AllClaims:
        DaysInactive: 60
        ExceptWhenOwnerHasTotalClaimBlocks: 10000
        ExceptWhenOwnerHasBonusClaimBlocks: 5000
      AutomaticNatureRestoration:
        SurvivalWorlds: false
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: true
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      world: true
      world_nether: true
      world_the_end: true
      spawn: true
    ProtectFreshSpawns: true
    PunishLogout: true
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: false
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: false
  Economy:
    ClaimBlocksMaxBonus: 0
    ClaimBlocksPurchaseCost: 0.0
    ClaimBlocksSellValue: 0.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: true
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: true
  LimitSkyTrees: true
  LimitTreeGrowth: false
  PistonMovement: CLAIMS_ONLY
  PistonExplosionSound: true
  FireSpreads: false
  FireDestroys: false
  AdminsGetWhispers: true
  AdminsGetSignNotifications: true
  VisualizationAntiCheatCompatMode: false
  SmartBan: true
  Mute New Players Using Banned Words: true
  MaxPlayersPerIpAddress: 3
  SilenceBans: true
  Siege:
    Worlds: []
    BreakableBlocks:
    - GRASS_BLOCK
    - DIRT
    - COBBLESTONE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - SAND
    - GRAVEL
    - GLASS
    - GRASS     <-- can't understand this entry, see BukkitDev documentation
    - FERN
    - DEAD_BUSH
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    - GLASS_PANE
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 90
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false
```

### Running without GriefPrevention

- [ ] I attempted running the server without GriefPrevention installed.
- [X] The problem does _not_ occur when GriefPrevention is removed from the server.

### Running with only GriefPrevention

- [ ] I attempted running only GriefPrevention on the server.
- [ ] The issue still occurs when GriefPrevention is the only plugin running.

### Running on a fresh, clean server installation

- [ ] I attempted testing for the issue on a new server.
- [ ] The issue still occurs on a new server.

### Using unmodified client

- [ ] I attempted testing for the issue with the vanilla client.
- [ ] The issue still occurs when using the vanilla client.

### We appreciate you taking the time to fill out a bug report!

- [X] I searched for similar issues before submitting this bug report.

**Comment by QarthO** [comment]
The error is saying "message can not be null"
Did you edit one of the messages in ``messages.yml`` and leave it blank?

Can you upload your ``messages.yml``?

Also, in your config, ``    - GRASS     <-- can't understand this entry, see BukkitDev documentation`` you will need to remove this since grass was changed to short_grass and gp doesnt support it yet

**Comment by Jikoo** [comment]
You claim to be running 16.18.1, but the stack trace indicates that you are running 16.18. This bug was fixed in #1880. Update to 16.18.1.
